include/circt/Dialect/LLHD/Simulator/Engine.h

```diff
@@ -13,6 +13,9 @@
 #ifndef CIRCT_DIALECT_LLHD_SIMULATOR_ENGINE_H
 #define CIRCT_DIALECT_LLHD_SIMULATOR_ENGINE_H
 
+#include "State.h"
```

Comment:
Why do we need this include? There are forward-declarations of `State` and `Instance` below, so shouldn't we be able to get rid of one of them?

Reply:
Thanks and I miss it, prefer to remove the declarations of `State` and `Instance`.